### PR TITLE
Replace unicode errors when decoding sst table

### PR DESCRIPTION
**Pull request by jendabenda**
Fixes xls with malformed sst table.

### CODE REVIEW

File: xlrd/book.py [hunk 1]
```diff
@@ -1452,7 +1452,7 @@ def unpack_SST_table(datatab, nstrings):
                 rawstrg = data[pos:pos+2*charsavail]
                 # if DEBUG: print "SST U16: nchars=%d pos=%d rawstrg=%r" % (nchars, pos, rawstrg)
                 try:
-                    accstrg += unicode(rawstrg, "utf_16_le")
+                    accstrg += unicode(rawstrg, "utf_16_le", 'replace')
```
Comment (piskvorky) [inline xlrd/book.py:1455]: How can such invalid encoding happen?

Is the whole encoding wrong (not necessarily `utf_16_le`), or just a local corruption?

Comment (jendabenda) [reply xlrd/book.py:1455]: There is high unicode codepoint which are represented as surrogate pair in utf16. It turned out that python does not allowe surrogates by default. So they need to be allowed explicitely. Must match to value encoding, like in https://github.com/RaRe-Technologies/pii_tools/pull/786

Comment (piskvorky) [reply xlrd/book.py:1455]: What do you mean? The encoding here and in #786 does not match, `utf_16_le` vs `utf8`.

Comment (jendabenda) [reply xlrd/book.py:1455]: Thats poorly written sentence. Encoding type itself can be arbitrary, but encoding process have to be configured in the same way (`surrogatepass`)

**Pull request merged**